### DISCUSSION TITLE
dwarfs-0.7.0-RC6

### DISCUSSION THREAD

**Discussion by mhx** [mhx/dwarfs#147] [General]
## Features

- Support delayed loading of WinFsp DLL for universal binary. This makes the `mkdwarfs`, `dwarfsck` and `dwarfsextract` tools of the universal binary usable without the WinFsp DLL.

## Performance

- Optimized the offset cache to improve random read latency as well as sequential read latency. This gave a [100x higher throughput](https://github.com/mhx/dwarfs/issues/142) for a case where DwarFS was used to compress raw file system images. Fixes github #142.

## Bugfixes

- Fixed building with `make` instead of `ninja`. Also fix builing in `Debug` mode. Fixes github #146.
- Fixed `ninja clean`.
- Fixed symlink creation for `mount.dwarfs`/`mount.dwarfs2`.

## Other

- Added [CI pipeline](https://github.com/mhx/dwarfs/actions).
- Don't write versioning files to source tree.

<hr /><em>This discussion was created from the release <a href='https://github.com/mhx/dwarfs/releases/tag/v0.7.0-RC6'>dwarfs-0.7.0-RC6</a>.</em>

**Comment by adminx01** [comment]
Very nice performance improvements!
